cpp/src/arrow/type.h

```diff
@@ -1535,6 +1535,22 @@ std::shared_ptr<Schema> schema(
 
 /// @}
 
+/// \brief Unifies schemas
```

Comment:
nit: would "MergeSchemas" be a better name? I don't have a strong opinion. You might try to summarize what the method does in a single sentence in the `\brief` block

Reply:
@bkietz: what's your preference for the name? since you came up with "UnifiySchemas"